libbeat/cfgfile/list.go

```diff
@@ -102,13 +102,13 @@ func (r *RunnerList) Reload(configs []*reload.ConfigWithMeta) error {
 	for hash, config := range startList {
 		runner, err := createRunner(r.factory, r.pipeline, config)
 		if err != nil {
-			if _, ok := err.(*common.ErrInputNotFinished); ok {
+			if _, ok := err.(*common.ErrInputNotFinished); ok { //nolint:errorlint // ErrInputNotFinished is a struct type, not an expression/error value
```

Comment:
So, you can do `errors.Is(err, &common.ErrInputNotFinished{})`, but I don't know if that'll produce the same result as a typecast, since I assume `errors.Is` is comparing the values, and not just the types?

Reply:
(long sigh)... well, the nice thing to do then would be to implement `Is` on `ErrInputNotFinished` since value-equality is probably not useful for this type of error (see https://pkg.go.dev/errors#Is). But that is arguably out of scope here so I'm ok with the lint workaround for now 😜

Reply:
Yah, it's definitely kinda awkward, I was just really hesitant to tinker around with some unrelated code that I've never touched, since that's bitten me before.